src/cryptonote_core/tx_pool.cpp

```diff
@@ -820,7 +820,7 @@ namespace cryptonote
     return true;
   }
   //---------------------------------------------------------------------------------
-  void tx_memory_pool::set_relayed(const epee::span<const crypto::hash> hashes, const relay_method method)
+  void tx_memory_pool::set_relayed(const epee::span<const crypto::hash> hashes, const relay_method method, std::vector<bool> &just_broadcasted)
```

Comment:
It doesn't check that `just_broadcasted` is empty, so it can `emplace_back` in the wrong places in the loop.
```
if (!just_broadcasted.empty())
    just_broadcasted.clear();
```

Reply:
Done